src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1349 @@ static ColliderCylinderInit D_80B37A5C = {
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35D90.s")
+    if ((this->fireTimer != 0) || ((this->actor.colorFilterTimer != 0) && (this->actor.colorFilterParams & 0x4000))) {
+        switch (limbIndex) {
+            case WOLFOS_LIMB_EYES:
+                bodyPartIndex = 0;
+                break;
+            case WOLFOS_LIMB_FRONT_RIGHT_LOWER_LEG:
+                bodyPartIndex = 1;
+                break;
+            case WOLFOS_LIMB_FRONT_LEFT_LOWER_LEG:
+                bodyPartIndex = 2;
+                break;
+            case WOLFOS_LIMB_THORAX:
+                bodyPartIndex = 3;
+                break;
+            case WOLFOS_LIMB_ABDOMEN:
+                bodyPartIndex = 4;
+                break;
+            case WOLFOS_LIMB_TAIL:
+                bodyPartIndex = 5;
+                break;
+            case WOLFOS_LIMB_BACK_RIGHT_SHIN:
+                bodyPartIndex = 6;
+                break;
+            case 37:
+                //! @bug There is no limb with index this large, so bodyPartsPos[7] is uninitialised. Thus a flame will
+                //! be drawn at 0,0,0 when the Wolfos is on fire.
+                bodyPartIndex = 7;
+                break;
+            case WOLFOS_LIMB_BACK_RIGHT_PASTERN:
+                bodyPartIndex = 8;
+                break;
+            case WOLFOS_LIMB_BACK_LEFT_PAW:
+                bodyPartIndex = 9;
+                break;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35E4C.s")
+        if (bodyPartIndex >= 0) {
+            Vec3f bodyPartPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35EE4.s")
+            Matrix_MultVec3f(&bodyPartVec, &bodyPartPos);
+            this->bodyPartsPos[bodyPartIndex].x = bodyPartPos.x;
+            this->bodyPartsPos[bodyPartIndex].y = bodyPartPos.y;
+            this->bodyPartsPos[bodyPartIndex].z = bodyPartPos.z;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B360E8.s")
+static void* sWolfosNormalEyeTextures[] = { gWolfosNormalEyeOpenTex, gWolfosNormalEyeHalfTex, gWolfosNormalEyeNarrowTex,
+                                            gWolfosNormalEyeHalfTex };
+static void* sWolfosWhiteEyeTextures[] = { gWolfosWhiteEyeOpenTex, gWolfosWhiteEyeHalfTex, gWolfosWhiteEyeNarrowTex,
+                                           gWolfosWhiteEyeHalfTex };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B361A0.s")
+void EnWf_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnWf* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B36288.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_wf.c", 2157);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B36328.s")
+    // This condition will always return true, since unk_300 is true whenever action is WOLFOS_ACTION_WAIT_TO_APPEAR.
```

Comment:
🧐 👉 📓 
```suggestion
    // This conditional will always evaluate to true, since unk_300 is false whenever action is WOLFOS_ACTION_WAIT_TO_APPEAR.
```